packages/@internationalized/date/src/string.ts

```diff
@@ -22,6 +22,10 @@ const DATE_RE = /^(\d{4})-(\d{2})-(\d{2})$/;
 const DATE_TIME_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?$/;
 const ZONED_DATE_TIME_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?(?:([+-]\d{2})(?::(\d{2}))?)?\[(.*?)\]$/;
 const ABSOLUTE_RE = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?(?:(?:([+-]\d{2})(?::(\d{2}))?)|Z)$/;
+const DATE_TIME_DURATION_RE =
+    /^(?<negative>-)?P((?<years>-?\d*)Y)?((?<months>-?\d*)M)?((?<weeks>-?\d*)W)?((?<days>-?\d*)D)?((?<time>T)((?<hours>-?\d*[.,]?\d{1,9})H)?((?<minutes>-?\d*[.,]?\d{1,9})M)?((?<seconds>-?\d*[.,]?\d{1,9})S)?)?$/;
```

Comment:
I guess the regex doesn't need to include the minus sign for each component since none of them allow negative numbers, but maybe this offers better error messages anyway.

Reply:
Will make this update to remove the `-` for each component.

Reply:
Removed the `-` from each component.